File: shared_resources/src/main/java/com/thewizrd/shared_resources/utils/NumberUtils.java
package com.thewizrd.shared_resources.utils;

import androidx.annotation.Nullable;

public class NumberUtils {
    @Nullable
    public static Integer tryParseInt(String number) {
        Integer result = null;
        try {
            result = Integer.parseInt(number);
        } catch (NumberFormatException | NullPointerException ignored) {
        }

        return result;
    }

    public static Integer tryParseInt(String number, int defaultValue) {
        Integer result = null;
        try {
            result = Integer.parseInt(number);
        } catch (NumberFormatException | NullPointerException ignored) {
        }

        if (result != null)
            return result;
        else
            return defaultValue;
    }

    @Nullable
    public static Float tryParseFloat(String number) {
        Float result = null;
        try {
            result = Float.parseFloat(number);
        } catch (NumberFormatException | NullPointerException ignored) {
        }

        return result;
    }

    public static Float tryParseFloat(String number, float defaultValue) {
        Float result = null;
        try {
            result = Float.parseFloat(number);
        } catch (NumberFormatException | NullPointerException ignored) {
        }

        if (result != null)
            return result;
        else
            return defaultValue;
    }

    @Nullable
    public static Double tryParseDouble(String number) {
        Double result = null;
        try {
            result = Double.parseDouble(number);
        } catch (NumberFormatException | NullPointerException ignored) {
        }

        return result;
    }

    public static Double tryParseDouble(String number, double defaultValue) {
        Double result = null;
        try {
            result = Double.parseDouble(number);
        } catch (NumberFormatException | NullPointerException ignored) {
        }

        if (result != null)
            return result;
        else
            return defaultValue;
    }

    public static String toString(Float number) {
        if (number == null) {
            return null;
        }
        return number.toString();
    }

    public static String toString(Integer number) {
        if (number == null) {
            return null;
        }
        return number.toString();
    }

    public static Integer getValueOrDefault(Integer number, int defaultValue) {
        if (number == null)
            return defaultValue;

        return number;
    }

    public static Float getValueOrDefault(Float number, float defaultValue) {
        if (number == null)
            return defaultValue;

        return number;
    }

    public static Double getValueOrDefault(Double number, double defaultValue) {
        if (number == null)
            return defaultValue;

        return number;
    }
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/utils/NumberUtils.kt
package com.thewizrd.shared_resources.utils

object NumberUtils {
    @JvmStatic
    fun String.tryParseInt(): Int? {
        return this.toIntOrNull()
    }

    @JvmStatic
    fun String.tryParseInt(defaultValue: Int): Int {
        return this.toIntOrNull() ?: defaultValue
    }

    @JvmStatic
    fun String.tryParseFloat(): Float? {
        return this.toFloatOrNull()
    }

    @JvmStatic
    fun String.tryParseFloat(defaultValue: Float): Float {
        return this.toFloatOrNull() ?: defaultValue
    }

    @JvmStatic
    fun String.tryParseDouble(): Double? {
        return this.toDoubleOrNull()
    }

    @JvmStatic
    fun String.tryParseDouble(defaultValue: Double): Double {
        return this.toDoubleOrNull() ?: defaultValue
    }

    @JvmStatic
    fun toString(number: Float?): String? {
        return number?.toString()
    }

    @JvmStatic
    fun toString(number: Int?): String? {
        return number?.toString()
    }

    @JvmStatic
    fun Int?.getValueOrDefault(defaultValue: Int): Int {
        return this ?: defaultValue
    }

    @JvmStatic
    fun Float?.getValueOrDefault(defaultValue: Float): Float {
        return this ?: defaultValue
    }

    @JvmStatic
    fun Double?.getValueOrDefault(defaultValue: Double): Double {
        return this ?: defaultValue
    }
}